include/swift/Frontend/Frontend.h

```diff
@@ -213,9 +214,14 @@ class CompilerInvocation {
   /// Computes the runtime resource path relative to the given Swift
   /// executable.
   static void computeRuntimeResourcePathFromExecutablePath(
-      StringRef mainExecutablePath,
+      StringRef mainExecutablePath, bool shared,
       llvm::SmallString<128> &runtimeResourcePath);
 
+  /// Appends `lib/swift[_static]` to the given path
+  static void appendLibDir(llvm::SmallString<128> &path, bool shared);
```

Comment:
I would say that `SmallVectorImpl<char>` is better than the hardcoded stack size.